demisto_sdk/commands/common/files/file.py

```diff
@@ -0,0 +1,373 @@
+import shutil
+import urllib.parse
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from io import BytesIO
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+import requests
+from bs4.dammit import UnicodeDammit
+from pydantic import BaseModel, PrivateAttr, validator
+from requests.exceptions import RequestException
+
+from demisto_sdk.commands.common.constants import (
+    DEMISTO_GIT_PRIMARY_BRANCH,
+    DEMISTO_GIT_UPSTREAM,
+    urljoin,
+)
+from demisto_sdk.commands.common.files.errors import (
+    FileContentReadError,
+    FileReadError,
+    FileWriteError,
+    GitFileReadError,
+    HttpFileReadError,
+    LocalFileReadError,
+)
+from demisto_sdk.commands.common.git_content_config import GitContentConfig
+from demisto_sdk.commands.common.git_util import GitUtil
+from demisto_sdk.commands.common.handlers.xsoar_handler import XSOAR_Handler
+from demisto_sdk.commands.common.logger import logger
+
+
+class File(ABC, BaseModel):
+    git_util: GitUtil
+    input_path: Path
+    _input_path_content: bytes = PrivateAttr(None)
+    default_encoding: str = "utf-8"  # default encoding is utf-8
+
+    class Config:
+        arbitrary_types_allowed = (
+            True  # allows having custom classes for properties in model
+        )
+
+    @validator("git_util", pre=True, always=True)
+    def get_git_util(cls, v: Optional[GitUtil]) -> GitUtil:
+        return v or GitUtil.from_content_path()
+
+    @validator("input_path", always=True)
+    def get_input_path(cls, v: Path, values: Dict) -> Path:
+        input_path = v
+        git_util = values["git_util"]
+
+        if input_path.is_absolute():
+            return input_path
+        else:
+            logger.debug(
+                f"path {input_path} is not absolute, trying to get full relative path from {git_util.repo.working_dir}"
+            )
+
+        input_path = git_util.repo.working_dir / input_path
+        if not input_path.exists():
+            raise FileNotFoundError(f"File {input_path} does not exist")
+
+        return input_path
+
+    @property
+    def input_file_content(self) -> bytes:
```

Comment:
Lets call it `content` and add a new method for `text`.
`content` will call `read_bytes` and `text` will call `read_text`

WDYT?

Reply:
but what if its not a text file? putting the text method in `File` is not relevant for **all** files.